### ISSUE TITLE
Support logical assignment

### ISSUE THREAD

**Issue by Kingwl** [chakra-core/ChakraCore#6658]
https://github.com/tc39/proposal-logical-assignment

**Comment by Kingwl** [comment]
May I take this one?

**Comment by rhuanjl** [comment]
Yes this would be good to have - though it may be good to finish one of your current PRs first - purely from a house keeping perspective.

This implementation of the coalescing operator may show you a bit of how to start if you need it: https://github.com/chakra-core/ChakraCore/pull/6545
